storagemarket/impl/clientstates/client_fsm.go

```diff
@@ -137,70 +154,96 @@ var ClientEvents = fsm.Events{
 				deal.PollErrorCount++
 			}
 			deal.Message = fmt.Sprintf("Provider state: %s", storagemarket.DealStates[providerState])
+			switch storagemarket.DealStates[providerState] {
+			case "StorageDealVerifyData":
+				deal.AddLog("provider is verifying the data")
+			case "StorageDealPublish":
+				deal.AddLog("waiting for provider to publish the deal on-chain") // TODO: is that right?
+			case "StorageDealPublishing":
+				deal.AddLog("provider has submitted the deal on-chain") // TODO: is that right?
+			case "StorageDealProviderFunding":
+				deal.AddLog("waiting for provider to lock collateral on-chain") // TODO: is that right?
+			default:
+				deal.AddLog(deal.Message)
+			}
 			return nil
 		}),
 	fsm.Event(storagemarket.ClientEventResponseDealDidNotMatch).
 		From(storagemarket.StorageDealCheckForAcceptance).To(storagemarket.StorageDealFailing).
 		Action(func(deal *storagemarket.ClientDeal, responseCid cid.Cid, proposalCid cid.Cid) error {
 			deal.Message = xerrors.Errorf("miner responded to a wrong proposal: %s != %s", responseCid, proposalCid).Error()
+			deal.AddLog(deal.Message)
 			return nil
 		}),
 	fsm.Event(storagemarket.ClientEventDealRejected).
 		From(storagemarket.StorageDealCheckForAcceptance).To(storagemarket.StorageDealFailing).
 		Action(func(deal *storagemarket.ClientDeal, state storagemarket.StorageDealStatus, reason string) error {
 			deal.Message = xerrors.Errorf("deal failed: (State=%d) %s", state, reason).Error()
+			deal.AddLog(deal.Message)
 			return nil
 		}),
 	fsm.Event(storagemarket.ClientEventDealAccepted).
 		From(storagemarket.StorageDealCheckForAcceptance).To(storagemarket.StorageDealProposalAccepted).
 		Action(func(deal *storagemarket.ClientDeal, publishMessage *cid.Cid) error {
 			deal.PublishMessage = publishMessage
 			deal.Message = ""
+			deal.AddLog("deal has been accepted")
 			return nil
 		}),
 	fsm.Event(storagemarket.ClientEventStreamCloseError).
 		FromAny().To(storagemarket.StorageDealError).
 		Action(func(deal *storagemarket.ClientDeal, err error) error {
 			deal.Message = xerrors.Errorf("error attempting to close stream: %w", err).Error()
+			deal.AddLog(deal.Message)
 			return nil
 		}),
 	fsm.Event(storagemarket.ClientEventDealPublishFailed).
 		From(storagemarket.StorageDealProposalAccepted).To(storagemarket.StorageDealError).
 		Action(func(deal *storagemarket.ClientDeal, err error) error {
 			deal.Message = xerrors.Errorf("error validating deal published: %w", err).Error()
+			deal.AddLog(deal.Message)
 			return nil
 		}),
 	fsm.Event(storagemarket.ClientEventDealPublished).
 		From(storagemarket.StorageDealProposalAccepted).To(storagemarket.StorageDealAwaitingPreCommit).
 		Action(func(deal *storagemarket.ClientDeal, dealID abi.DealID) error {
 			deal.DealID = dealID
+			deal.AddLog(deal.Message)
```

Comment:
Does `deal.Message` have a value here?

Reply:
I don't think so, however we definitely want to call `AddLog` so that we update the timestamps for the stage. Changing to "".